tests/integration/test_fenicsprecice.py

```diff
@@ -207,103 +207,81 @@ def test_update_expression_vector(self):
-#         for ps in y_forces:
-#             ps.apply(b_adapter)
-#
-#         solve(a, u_manual.vector(), b_manual)
-#         solve(a, u_adapter.vector(), b_adapter)
-#
-#         assert(u_manual == u_adapter)
+def clamped_boundary(x, on_boundary):
+    return on_boundary and abs(x[1]) < 1E-14
+
+
+@patch.dict('sys.modules', **{'precice': MockedPrecice})
+class TestPointSource(TestCase):
+    """
+    Test Point Source return mechanism for force vector values given by user.
+    """
+    dummy_config = "tests/precice-adapter-config.json"
+
+    mesh = UnitSquareMesh(10, 10)
+    V = VectorFunctionSpace(mesh, "P", 2)
+
+    def test_get_point_sources_2d(self):
+        """
+        Checks
+        Returns
+        -------
+        """
+        from precice import Interface
+        import fenicsprecice
+        from fenicsprecice.adapter_core import FunctionType, filter_point_sources
+
+        n_vertices = 11
+        dimensions = 2
+
+        vertices_x = [1 for _ in range(n_vertices)]
+        vertices_y = np.linspace(0, 1, n_vertices)
+        vertices = []
+        for i in range(n_vertices):
+            vertices.append([vertices_x[i], vertices_y[i]])
+        vertices = np.array(vertices)
+
+        fixed_boundary = AutoSubDomain(clamped_boundary)
+
+        precice = fenicsprecice.Adapter(self.dummy_config)
+        precice._read_function_space = self.V
+        precice._Dirichlet_Boundary = AutoSubDomain(clamped_boundary)
+        precice._read_function_type = FunctionType.VECTOR
+
+        # Define 2D dummy forces
+        dummy_nodal_data = []
+        for i in range(n_vertices):
+            dummy_nodal_data.append([i, i])
+
+        dummy_nodal_data = np.array(dummy_nodal_data)
+
+        data = {tuple(key): value for key, value in zip(vertices, dummy_nodal_data)}
+
+        dummy_forces = []
+        for d in range(dimensions):
+            dummy_forces.append(dict())
+
+        for i in range(n_vertices):
+            key = []
+            for d in range(dimensions):
+                key.append(vertices[i, d])
+            key = tuple(key)
+
+            for d in range(dimensions):
+                dummy_forces[d][key] = PointSource(self.V.sub(d), Point(key), dummy_nodal_data[i, d])
+
+        # Avoid application of PointSource and Dirichlet boundary condition at the same point by filtering
+        for d in range(dimensions):
+            dummy_forces[d] = filter_point_sources(dummy_forces[d], fixed_boundary, warn_duplicate=False)
+
+        print("dummy_forces X: {}".format(dummy_forces[0].values()))
+        print("dummy_forces Y: {}".format(dummy_forces[1].values()))
+
+        # Get forces from Adapter function
+        forces_x, forces_y = precice.get_point_sources(data)
+
+        print("forces X: {}".format(forces_x))
+        print("forces Y: {}".format(forces_y))
+
+        assert(np.allclose(dummy_forces[0].values(), forces_x))
+        assert(np.allclose(dummy_forces[1].values(), forces_y))
```

Comment:
@BenjaminRodenberg there is a problem here. The function `get_point_sources` returns a list of PointSource objects. Even if we generate the same PointSource objects it is not possible to check if they carry the same forces as we cannot simple compare objects. Any ideas how we can test this?

Also I think this particular test and also some tests above are in fact unit tests and we keep them in `tests/integration`, which is confusing.